aicsimageio/aics_series.py

```diff
@@ -0,0 +1,367 @@
+        self._shape = None
+        self._size = None
+
+    @property
+    def series_dim(self) -> str:
+        """
+        Returns
+        -------
+        series_dim: str
+            Which dimension is supposed to act like a normal array axis but is actually reading a series of images.
+        """
+        return self._series_dim
+
+    @property
+    def operating_index(self) -> int:
+        """
+        Returns
+        -------
+        operating_index: int
+            The dimension index of the series dimension.
+        """
+        return constants.DEFAULT_DIMENSION_ORDER.index(self.series_dim)
+
+    @property
+    def images(self) -> List[Path]:
+        """
+        Returns
+        -------
+        images: List[Path]:
+            The list of filepaths to images to be used for the series.
+        """
+        return self._images
+
+    @staticmethod
+    def _ensure_valid_data_shape(
+        data_shape: Tuple[int],
+        operating_index: int,
+        prior_shape: Optional[Tuple[int]] = None
+    ):
+        """
+        Used to ensure that the data shape is `1` in whichever index is the operating index for the series.
+        Additionally, if provided a prior shape, will check for shape consistenting between the pair.
+
+        Raises an InvalidDimensionOrderingError or InconsitentDataShapeException.
+
+        Parameters
+        ----------
+        data_shape: Tuple[int]
+            A tuple of six dimension sizes produced by `AICSImage.size()` for an image in the series.
+        operating_index: int
+            The operating index for the series. This is constants.DEFAULT_DIMENSION_ORDER.index(self.series_dim).
+        prior_shape: Optional[Tuple[int]] = None
+            An optional prior shape to ensure that the provided data shape is consistent with the prior.
+        """
+        if data_shape[operating_index] != 1:
+            raise exceptions.InvalidDimensionOrderingError(
+                f"The read data shape is invalid for the current operating series dimension. Read shape: {data_shape}."
+            )
+
+        if prior_shape:
+            if data_shape != prior_shape:
+                raise exceptions.InconsitentDataShapeException(
+                    f"The read data shape is inconsitent with the prior data shape. "
+                    f"Read shape: {data_shape}, prior shape: {prior_shape}"
+                )
+
+    def size(self, dims: str = constants.DEFAULT_DIMENSION_ORDER):
+        """
+        Parameters
+        ----------
+        dims: A string containing a list of dimensions being requested.  The default is to return the 6 standard dims
+
+        Returns
+        -------
+        Returns a tuple with the requested dimensions filled in
+        """
+        # Only run if shape has never been retrieved before
+        if self._size is None:
+            with AICSImage(self.images[0]) as img:
+                # Get size
+                shape = img.size()
```

Comment:
when i see `shape = img.size()` it makes me think something is named wrongly :)

Reply:
I think it's because I now just fully think in terms of numpy array method names and attributes. I think it being called `size` though is accurate. It is the images size, but that also just happens to be the ndarray shape.